packages/web/src/types.ts

```diff
@@ -0,0 +1,281 @@
+  approvalProgram: string
+  clearProgram: string
+  localInts: number
+  localBytes: number
+  globalInts: number
+  globalBytes: number
+  approvalProg?: Uint8Array
+  clearProg?: Uint8Array
+};
+
+export type UpdateAppParam = BasicParams & AppOptionalFlags & {
+  type: TransactionType.UpdateApp
+  appID: number
+  newApprovalProgram: string
+  newClearProgram: string
+  approvalProg?: Uint8Array
+  clearProg?: Uint8Array
+};
+
+export type OptInAppParam = BasicParams & AppOptionalFlags & {
+  type: TransactionType.OptInApp
+  appID: number
+};
+
+export type OptInASAParam = BasicParams & {
+  type: TransactionType.OptInASA
+  assetID: number | string
+};
+
+export type ModifyAssetParam = BasicParams & {
+  type: TransactionType.ModifyAsset
+  assetID: number | string
+  fields: AssetModFields
+};
+
+export type FreezeAssetParam = BasicParams & {
+  type: TransactionType.FreezeAsset
+  assetID: number | string
+  freezeTarget: AccountAddress
+  freezeState: boolean
+};
+
+export type RevokeAssetParam = BasicParams & {
+  type: TransactionType.RevokeAsset
+  /**
+   * Revoked assets are sent to this address
+   */
+  recipient: AccountAddress
+  assetID: number | string
+  /** Revocation target is the account from which the clawback revokes asset. */
+  revocationTarget: AccountAddress
+  amount: number | bigint
+};
+
+export type DestroyAssetParam = BasicParams & {
+  type: TransactionType.DestroyAsset
+  assetID: number | string
+};
+
+export type AlgoTransferParam = BasicParams & {
+  type: TransactionType.TransferAlgo
+  toAccountAddr: AccountAddress
+  amountMicroAlgos: number | bigint
+};
+
+export type AssetTransferParam = BasicParams & {
+  type: TransactionType.TransferAsset
+  toAccountAddr: AccountAddress
+  amount: number | bigint
+  assetID: number | string
+};
+
+export type AppCallsParam = BasicParams & AppOptionalFlags & {
+  type: TransactionType.CallNoOpSSC | TransactionType.ClearApp |
+  TransactionType.CloseApp | TransactionType.DeleteApp
+  appID: number
+};
+
+// Algosigner types
+
+export interface AlgoSignerTx {
+  "consensus-version": string
+  "fee": number
+  "genesis-hash": string
+  "genesis-id": string
+  "last-round": number
+  "min-fee": number
+}
+
+export interface AlgoSignerSignedTx {
+  "txID": string
+  "blob": string
+}
+
+export interface AlgoSignerSendTx {
+  "txId": string
+}
+
+export interface AlgoSignerToBeSignedTx {
+  txn: string
+}
+
+export interface encoding {
+  msgpackToBase64: (txn: Uint8Array) => string
+
+  base64ToMsgpack: (txn: string) => Uint8Array
+}
+
+export interface AlgoSigner {
+  // algod, sign, send, signTxn, accounts
+  // Create an Algod client to get suggested transaction params
+  // let client = new algosdk.Algodv2(token, server, port, headers);
+  accounts: (param: unknown) => unknown[]
+
+  algod: (param: unknown) => any
+
+  signTxn: (param: unknown[]) => AlgoSignerSignedTx
+
+  send: (param: any) => AlgoSignerSendTx
+
+  encoding: encoding
+}
```

Comment:
Algosigner is written in typescript. Can we use their typings?

Reply:
yes, but we cannot import these here, will have to copy paste.
i will add a task for it, in integration.